test/server/helpers/assert-error.js

```diff
@@ -63,7 +63,14 @@ function assertThrow (fn, expectedErr) {
         actualErr = err;
     }
 
-    expect(actualErr).eql(expectedErr);
+    const keys = Object.keys(actualErr);
+
+    for (const key of keys) {
+        //NOTE: we skip the originError because it's APIError object
+        if (key !== 'originError')
```

Comment:
I would prefer to initialize `originError` with `null` in the test data, but understand that it can be a lot of work. Let's add a TODO comment instead of the note.